### ISSUE TITLE
add option to produce css that doesn't require crazy escaping

### ISSUE THREAD

**Issue by ahdinosaur** [mmckegg/micro-css#1]
at the moment the input

```
Thing {
  .name {
    color: green;
  } 
}

$green {
  color: green;
}
```

produces:

```
.Thing.\.name {
  color: green;
}

.\$green {
  color: green;
}
```

where as a querySelector it's `.Thing.\\.name` and `.\\$green`.

this is because `.` and `$` are special characters that must be [escaped](https://mathiasbynens.be/notes/css-escapes).

i wonder if it's possible to configure these characters in `micro-css` so that we don't need to do crazy escaping.

/cc @mmckegg @mixmix

**Comment by ahdinosaur** [comment]
also, in particular `.` is very confusing because anyone who stumbles on a `className` of `.name` might assume that it can be styled and queried with `.name`, when really it's `.\\.name`.

**Comment by mmckegg** [comment]
Yep, I mostly agree. I've been struggling for an alternative that still looks as nice in the html.

Any suggestions? @ahdinosaur @mixmix

**Comment by ahdinosaur** [comment]
i'd be happy with no prefix on `.` and `$` be changed to `_` which doesn't need to be escaped.

**Comment by mmckegg** [comment]
@ahdinosaur yeah that's a good idea! I was just thinking about that myself. That sounds very workable. The only "catch" would be potential class conflicts (i.e. some non-mcss interferring), but that has always been a problem with css, and is pretty easy to debug.

I'm definitely in support of this!

**Comment by mmckegg** [comment]
**Idea for backwards compat:** generate both styles for now. But deprecate the prefixes. Also, obviously bumping major.

**Comment by mmckegg** [comment]
okay this is done! [Released as v1.0](https://github.com/mmckegg/micro-css#breaking-changes-in-v10)

Decided not to bother with backwards compat because this is only a problem if you are not using the `micro-css/h` helper. And I don't think there are very many people using micro-css besides us!
